### ISSUE TITLE
vulkaninfo throws a warning

### ISSUE THREAD

**Issue by buzmeg** [KhronosGroup/MoltenVK#1517]
When I install the latest VulkanSDK, I get:

```
% vulkaninfo
WARNING: [MoltenVK] Code 0 : Unrecognized CreateInstance->pCreateInfo->pApplicationInfo->apiVersion number (0x004020c6). Assuming MoltenVK 1.1 version.
==========
VULKANINFO
==========

Vulkan Instance Version: 1.2.198
```

MacBook Pro 13" 2020  macOS Monterey 12.0.1

Thanks.

**Comment by billhollings** [comment]
`apiVersion  = 0x004020c6` indicates that your app requested Vulkan 1.2.198.

MoltenVK supports Vulkan 1.1, but does not support Vulkan 1.2 yet.

**Comment by buzmeg** [comment]
It's nice to know that I can ignore this.

However, vulkaninfo is part of the official Vulkan download for macOS from LunarG so this is not *my* app.

I strongly suspect that you have better contact with the developers responsible for that code than I do.

Thanks.

**Comment by billhollings** [comment]
> I strongly suspect that you have better contact with the developers responsible for that code than I do.

I've reported this to the Vulkan SDK [issue list](https://vulkan.lunarg.com/issue/view/620284935df1129bc631f6dc).

**Comment by billhollings** [comment]
> This is not a vulkaninfo issue.
> 
> An application can specify any instance version, as the apiVersion corresponds to the 'highest supported version the application can support'.
> 
> > apiVersion must be the highest version of Vulkan that the application is designed to use, encoded as described in https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#extendingvulkan-coreversions-versionnumbers.
> 
> https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkApplicationInfo.html

Hmmm...the next line of the Vulkan spec after the one you quote specifies:

>The patch version number specified in apiVersion is ignored when creating an instance object. Only the major and minor versions of the instance **_must match those requested in apiVersion_** [my empahsis].

So it seems to read like this value should be validated against what the driver can support. The spec is clear that no error should be returned, which is why MoltenVK does not do that, and just issues a warning. But we can remove that too if it is confusing, or overkill.

**Comment by charles-lunarg** [comment]
Ah, right. The application must not create an instance greater than what the *instance* can support. At this point in the application's lifetime, it has no way of knowing what the driver's support, only the instance. Thus, Vulkaninfo will set the apiVersion to the major.minor that it received from `vkEnumerateInstanceVersion`. I believe what is happening here is that the vulkan-loader on the system advertises support for 1.2.<its patch version>. 

The vulkaninfo version selection logic is found here: https://github.com/KhronosGroup/Vulkan-Tools/blob/7087fbbc85fbd5d47dd5fd4fef3c0cd35a52fdd1/vulkaninfo/vulkaninfo.h#L616

**Comment by billhollings** [comment]
> I believe what is happening here is that the vulkan-loader on the system advertises support for 1.2., and vulkaninfo just uses that.

Okay. Are you indicating that this should be handled within `vulkaninfo` after all?

**Comment by charles-lunarg** [comment]
No, because there is no way to know what the version of the drivers are before creating an instance. The only version it can query is `vkEnumerateInstanceVersion` which is the version of the loader on the system. If an application directly linked to moltenVK, then that function should return 1.1, since its the version moltenVK supports.

So I believe that MoltenVK shouldn't be reporting this warning. It is possible that an application could setup swiftshader (which supports 1.2 iirc) and thus would want to set 1.2 as the instance apiVersion.

**Comment by billhollings** [comment]
> So I believe that MoltenVK shouldn't be reporting this warning.

Okay. Makes sense. I'll track this as a bug here and remove the warning. 

And regarding the SDK issue, if you are satisfied all is good on the Loader/Layer operation on this, feel free to close that issue.

**Comment by billhollings** [comment]
Fixed in PR #1523. Closing this now.
